File: lua/bench.lua
local abc = require "lua.abc"

local it = abc.Model()

it.quacks = it.quacks
it.flies = it.flies
it.swims = it.swims
it.croaks = it.croaks

for i = 1, 1e5 do
    it.duck[i] = it.quacks * (it.flies + it.swims)
end

local tick = abc.Build(it)

local start = os.clock()

local state = tick()

print("compilation time: ", start)
print("calculation time: ", os.clock()-start)

File: python/bench.py
import abc_dsl as abc

import time

start = time.time()

it = abc.Model()

it.quacks = it.quacks
it.flies = it.flies
it.swims = it.swims
it.croaks = it.croaks

for i in range(1_000_000):
    it.duck[i] = it.quacks & (it.flies | it.swims)

tick, state = abc.Build(it)

print("compilation time: ", time.time() - start)

tick()

start = time.time()

tick()

print("calculation time: ", time.time()-start)